services/search/pkg/search/search.go

```diff
@@ -155,3 +156,31 @@ func convertToWebDAVPermissions(isShared, isMountpoint, isDir bool, p *provider.
 	}
 	return b.String()
 }
+
+func extractScope(path string) (*searchmsg.Reference, error) {
+	ref, err := storagespace.ParseReference(path)
+	if err != nil {
+		return nil, err
+	}
+	return &searchmsg.Reference{
+		ResourceId: &searchmsg.ResourceID{
+			StorageId: ref.ResourceId.StorageId,
+			SpaceId:   ref.ResourceId.SpaceId,
+			OpaqueId:  ref.ResourceId.OpaqueId,
+		},
+		Path: ref.GetPath(),
+	}, nil
+}
+
+// ParseScope extract a scope value from the query string
+func ParseScope(query string) (search, scope string) {
+	match := scopeRegex.FindStringSubmatch(query)
```

Comment:
How about bringing `scopeRegex` to this file? It's only used in here after all.

Reply:
done